include/flyft/data_layout.h

```diff
@@ -3,24 +3,39 @@
 
 #include <iterator>
 #include <type_traits>
+#include <vector>
 
 namespace flyft
     {
+//! DataLayout
+/*!
+ * Generates data structure for the custom data types used in the software.
+ * The class provides get and set functions for the array index and implements N-dimensional array
+ * mapping into one dimensional array using strided indexing scheme.
+ */
 class DataLayout
     {
     public:
     DataLayout();
-    explicit DataLayout(int shape_);
+    //! Constructor for the DataLayout
+    /*!
+     * \param shape_ Shape of the index array
+     */
```

Comment:
Don't include class name in short descriptions. Use periods to end parameter descriptions. Parameter name needs to match argument name.
```suggestion
    //! Constructor
    /*!
     * \param shape Number of elements in array along each dimension.
     */
```